src/main/java/io/github/crucible/grimoire/patch/GrimPatch.java

```diff
@@ -5,73 +5,106 @@
 import java.util.jar.Manifest;
 import java.util.zip.ZipEntry;
 
+import javax.annotation.Nullable;
+
 public class GrimPatch implements Comparable<GrimPatch> {
 
     private final long priority;
     private final String patchName;
     private final String modId;
     private final String targetJar;
     private final boolean corePatch;
-    private final List<ZipEntry> mixinEntries;
-    private final String fileName;
-
-    public GrimPatch(Manifest manifest, List<ZipEntry> mixinEntries, File modFile) {
-        this.priority       = getLong(manifest,"GRIMOIRE_PRIORITY", 0L);
-        this.patchName      = getString(manifest,"GRIMOIRE_PATCHNAME", modFile.getName());
-        this.modId          = getString(manifest,"GRIMOIRE_MODID", "");
-        this.targetJar      = getString(manifest,"GRIMOIRE_TARGETJAR", "");
-        this.corePatch      = getBoolean(manifest,"GRIMOIRE_COREPATCH", false);
-        this.mixinEntries   = mixinEntries;
-        this.fileName       = modFile.getName();
+    private final List<ZipEntry> mixinZipEntries;
+    private final List<File> mixinFileEntries;
+    private final String fileOrDirectoryName;
+
+    public GrimPatch(Manifest manifest, List<ZipEntry> mixinZipEntries, File modFile) {
+        this(manifest, mixinZipEntries, null, modFile);
+    }
+
+    public GrimPatch(Manifest manifest, File modFile, List<File> mixinFileEntries) {
```

Comment:
It might be better to just use List<String> for mixin entries. Mixin only needs the jar resource path for the mixin entry and I don't think it will even support a file.